### PR TITLE
When a list is scrolling, children can't be tapped

### LINKED ISSUE #5023: Stopping flings should not count as a touch
(A video would make this easy to understand, but it's not easy for me to capture one at this second.)

To repro:
1.  Open Flutter Gallery on iOS
2. Fling scroll on the home page.
3.  Stop the fling with your finger (notice how it starts a splash effect on the item below your finger).
4.  Lift your finger after stopping.  Notice that it counted your "stop scroll" as an action and activated the list item below your finger and took you to whatever demo you happened to be over.

Compare this behavior with how flings of other scrolling lists work on iOS (such as Settings), notice that Flutter's behavior is incorrect.

It's possible our behavior here is wrong on Android too, but I haven't confirmed yet.

Thanks to @goderbauer for bringing this to my attention.

### PR DESCRIPTION
**Pull request by Hixie**
Fixes https://github.com/flutter/flutter/issues/5023

### CODE REVIEW

**Comment by eseidelGoogle** [comment]
Thank you very much for fixing!

**Comment by HansMuller** [comment]
LGTM

**Comment by abarth** [comment]
LGTM BUT NOT TO TRAVIS

**Comment by Hixie** [comment]
The travis failure was because the gallery smoketest wasn't waiting for the scroll to settle before tapping the list, but that's now an ignored tap.

**Comment by Hixie** [comment]
@HansMuller you may be interested in the flutter gallery test change in this PR.

I'm going to check this in some time after Travis goes green unless I hear otherwise.

File: examples/flutter_gallery/test/smoke_test.dart [hunk 1]
```diff
@@ -43,8 +43,9 @@ Future<Null> smokeDemo(WidgetTester tester, String routeName) async {
 
   await tester.tap(menuItem);
   await tester.pump(); // Launch the demo.
-  await tester
-      .pump(const Duration(seconds: 1)); // Wait until the demo has opened.
+  await tester.pump(const Duration(seconds: 1)); // Wait until the demo has opened.
+
+  expect(find.text('Flutter Gallery'), findsNothing);
```
Comment (eseidelGoogle) [inline examples/flutter_gallery/test/smoke_test.dart]: I feel like we've changed the case of `Flutter gallery` and Flutter Gallery at times, hopefully this is finding or not finding the right thing. :)

**Comment by Hixie** [comment]
This blocked touches on lists for much longer than expected. I'm not sure if we're missing a setState when the animation ends or if we're just letting the animation go for much longer than it should, but either way, I'm reverting this for now.

**Pull request merged**